tests/unit/test_retry.py

```diff
@@ -0,0 +1,184 @@
+from datetime import datetime, timedelta
+from os import getenv
+from typing import AsyncIterator
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+from jbi.queue import QueueItem, get_dl_queue
+from jbi.retry import retry_failed
+
+RETRY_TIMEOUT_DAYS = getenv("RETRY_TIMEOUT_DAYS", 7)
+
+
+def add_iter(obj):
+    mock = MagicMock()
+    mock.__aiter__.return_value = obj
+    return mock
+
+
+def iter_error():
+    mock = MagicMock()
+    mock.__aiter__.return_value = None
+    mock.__aiter__.side_effect = Exception("Throwing an exception")
+    return mock
+
+
+@pytest.mark.asyncio
+async def test_retry_empty_list(caplog):
+    retrieve = AsyncMock(return_value={})
+    get_dl_queue().retrieve = retrieve
+
+    metrics = await retry_failed()
+    retrieve.assert_called_once()
+    assert len(caplog.messages) == 0
+    assert metrics == {
+        "bug_count": 0,
+        "events_processed": 0,
+        "events_skipped": 0,
+        "events_failed": 0,
+        "bugs_failed": 0,
+    }
+
+
+@pytest.mark.asyncio
+async def test_retry_success(caplog, queue_item_factory):
+    exec_mock = MagicMock()
+    queue = get_dl_queue()
+    queue.retrieve = AsyncMock(
+        return_value={
+            1: add_iter(
+                [
+                    queue_item_factory(
+                        payload__bug__id=1, payload__event__time=datetime.now()
+                    )
+                ]
+            )
+        }
+    )
+    queue.done = AsyncMock()
+
+    metrics = await retry_failed(item_executor=exec_mock)
+    assert len(caplog.messages) == 0
+    queue.retrieve.assert_called_once()  # no logs should have been generated
```

Comment:
nit: I think the trailing comment goes to the line above

Reply:
Yup, got misaligned when switching to caplog